volatility/framework/layers/vmware.py

```diff
@@ -81,12 +80,20 @@ def _read_header(self) -> None:
                         self._context.object("vmware!unsigned int",
                                              offset = offset + name_len + 2 + (index * index_len),
                                              layer_name = self._meta_layer))
-                data = self._context.object("vmware!unsigned int",
+                data_len = flags & 0x3f
+
+                # TODO: Read special data sizes (signalling a longer data stream) properly instead of skipping them
+                if data_len in (62, 63):
+                    data_len = 4 if version == 0 else 8
+                    offset += 2 + name_len + (indicies_len * index_len) + 2 * data_len
```

Comment:
So this confuses me, since it looks like we'd be doing this twice?  Can you walk me through what's going on here please?

Reply:
I tried to clearify this a bit more in the next commit. Most tags have a `data_len` of 4 or 8, which are the regular cases and easy to handle. The layout looks as follows:

| flags | name_len | name | indices | data |
| - | - | - | - | - |
| 1 | 1 | `name_len` | 4 * `indices_len` | `data_len` |

Some tags however, have a `data_len` of 62 or 63, which indicate a longer datastream. The layout is than as follows: 

| flags | name_len | name | indices | data_size | data_mem_size | padding? | data |
| - | - | - | - | - | - | - | - |
| 1 | 1 | `name_len` | 4 * `indices_len` | 4 or 8 | 4 or 8 | 2 | `data_size` |

The difference between `data_size` and `data_mem_size` should be different in cases where the data is compressed (`data_size < data_mem_size`).

I based this mostly on [Volatility 2.x-code](https://github.com/volatilityfoundation/volatility/blob/master/volatility/plugins/addrspaces/vmware.py) and [the original parser of the same author](https://github.com/DrDonk/vmsnparser/blob/master/source/vmsn.py).